app/controllers/v1/facilities/ccp_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'will_paginate/array'
+class V1::Facilities::CcpController < FacilitiesController
+  # Provider supports the following query parameters:
+  # @param bbox - Bounding box in form "xmin,ymin,xmax,ymax" in Lat/Long coordinates
+  # @param services - Optional specialty services filter
+  def index
+    api_results = ppms_search.collect do |result|
+      PPMS::Provider.new(
+        result.attributes.transform_keys { |k| k.to_s.snakecase.to_sym }
+      )
+    end.uniq(&:id).paginate(pagination_params)
+
+    render_json(serializer, ppms_params, api_results, { include: [:specialties] })
+  end
+
+  def show
+    api_result = api.provider_info(params[:id])
+
+    services = api.provider_services(params[:id])
+    api_result.add_provider_service(services[0]) if services.present?
+
+    api_result = PPMS::Provider.new(api_result.attributes.transform_keys { |k| k.to_s.snakecase.to_sym })
+
+    render_json(serializer, ppms_params, api_result, { include: [:specialties] })
+  end
+
+  def specialties
+    api_results = api.specialties.collect do |result|
+      PPMS::Specialty.new(
+        result.transform_keys { |k| k.to_s.snakecase.to_sym }
+      )
+    end
+
+    render_json(PPMS::SpecialtySerializer, ppms_params, api_results)
+  end
+
+  private
+
+  def api
+    @api ||= Facilities::PPMS::Client.new
+  end
+
+  def pagination_params
+    hsh = super
+    page = Integer(hsh[:page] || 1)
+    per_page = Integer(hsh[:per_page] || 1)
+    total_entries = page * per_page + 1
+    hsh.compact.transform_values!(&:to_i)
+    hsh.merge(
+      total_entries: total_entries
+    )
+  end
+
+  def ppms_params
+    params.permit(
+      :address,
+      :page,
+      :per_page,
+      :type,
+      bbox: [],
+      services: []
+    )
+  end
+
+  def ppms_search
+    if ppms_params[:type] == 'provider' && ppms_params['services'] == ['261QU0200X']
```

Comment:
I can move that logic into a method and show, I feel it needs to be done this way because if you provide the type as a provider and it has that specific service it needs to use a different code path.